src/GtkUtilities.jl

```diff
@@ -44,7 +44,7 @@ include("panzoom.jl")
 using .PanZoom
 import .PanZoom: interior, fullview   # for extensions
 
-function Base.copy!{C<:Colorant}(ctx::CairoContext, img::AbstractArray{C})
+function Base.copy!{C<:Union{Colorant,Number}}(ctx::CairoContext, img::AbstractArray{C})
```

Comment:
I understand your point, but the method one really uses is `copy!(c::Canvas, A)`. Should one have to write this as `copy!(GtkUtilities.Canvas(c), A)`? That would make `copy!(c, A)` fail with a `MethodError`, and people are then unlikely to discover the subtle change that makes this work. It would not be wrong to file a bug against Gtk saying that this was a missing method, so from that perspective this is not type piracy; people should just write `using GtkUtilities` and consider `Gtk` an internal module for `GtkUtilities`.

We have lots of other examples of this. The purists don't want math to be defined on Color types, the pragmatists do. Hence ColorVectorSpace. No one is going to wrap all their mathematics with `Colormath(rgb1) + Colormath(rgb2)`.

Given your involvement in managing `Pkg` and METADATA, you tend to hear particularly about type piracy's problems (they are genuine, and truly awful), but you probably hear less about the success stories. Type piracy is nasty except when it's not.